resource-aggregate/service/grpcApi.go

```diff
@@ -155,6 +155,9 @@ func (r RequestHandler) UnpublishResourceLinks(ctx context.Context, request *com
 		log.Errorf("cannot publish resource links unpublished events: %v", err)
 	}
 	auditContext := commands.NewAuditContext(owner, "")
+
+	r.UnpublishResources(ctx, request, owner, events)
```

Comment:
Before the device signs off, it will unpublish all resources. Am I correct?
If that's true, this will lead to a performance bottleneck as we have to go through every published resource and cancel its pending requests and create a snapshot. Do we want to really do it? What if some device starts to unpublish all links before it signs off?

Reply:
Yes, we want to try to do it. It means that all clients that wait for pending commands will get cancelled and the cache will not contain the old resource.